packages/react-form-state/src/FormState.ts

```diff
@@ -70,16 +72,31 @@ export default class FormState<
 
   static getDerivedStateFromProps<T>(newProps: Props<T>, oldState?: State<T>) {
     const newInitialValues = newProps.initialValues;
+    const newExternalErrors = newProps.externalErrors || [];
 
     if (oldState == null) {
-      return createFormState(newInitialValues);
+      return createFormState(newInitialValues, newExternalErrors);
     }
 
     const oldInitialValues = initialValuesFromFields(oldState.fields);
```

Comment:
Not related to the PR but I feel `getDerivedStateFromProps` is not necessary in this component.
- Reinitialization can be achieved in `componentDidUpdate` with access to prevProps